cohort-evaluator/src/main/java/com/ibm/cohort/cql/evaluation/CqlContextFactory.java

```diff
@@ -66,16 +66,20 @@ public ContextCacheKey(
 
         @Override
         public boolean equals(Object o2) {
-            boolean isEqual = o2 != null && o2 instanceof ContextCacheKey;
```

Comment:
The instanceof check implies non-null.

The equals check will fail even if both "this" and o2 are null, but as that is the current behavior I choose not to alter that.  Happy to change if desired.

Reply:
This method really can't handle when `this` is null since it's an instance method.
If we're in an instance method, then `this` is guaranteed to be non-null.